index.js

```diff
@@ -305,8 +314,13 @@ function asPromise(opts) {
 
 			onCancel(() => {
```

Comment:
I just noticed an existing issue. `p-cancelable` was never designed for multiple `onCancel` handlers, but we call it twice. Here and on line 306. So the first `onCancel` handler is ignored. I've fixed this in https://github.com/sindresorhus/got/commit/871c3bd25442edd3cd89190c3d4788c0efd5c2db